libnavigation-core/src/main/java/com/mapbox/navigation/core/location/replay/ReplayRouteLocationConverter.kt

```diff
@@ -52,6 +51,9 @@ internal class ReplayRouteLocationConverter(
 
     override fun setRoute(route: DirectionsRoute) {
         this.route = route
+        currentLeg = 0
+        currentStep = 0
+        time = 0
```

Comment:
Is this also a problem in the legacy 🤔? How are we handling it in the legacy implementation? https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/location/replay/ReplayRouteLocationConverter.kt

Reply:
no, it's in new approach only.
Legacy recreate converter each time on new route has been set, no options to `reset` route(take a look a [constructor](https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/location/replay/ReplayRouteLocationConverter.kt#L12))
In new approach no need to recreate converter, only reset cache-values when new route is set

Reply:
Thanks for confirming 🙇